setup.py

```diff
@@ -27,6 +29,23 @@
 else:
     raise RuntimeError("Unable to find version string in %s." % (VERSIONFILE,))
 
+# Dependencies of GPflowOpt
+dependencies = ['numpy>=1.9', 'scipy>=0.16', 'GPflow>=0.3.5']
+min_tf_version = '1.0.0'
+
+# Detect if TF is installed or outdated.
+# If the right version is installed, do not list as requirement to avoid installing over e.g. tensorflow-gpu
+# To avoid this, rely on importing rather than the package name (like pip).
+try:
+    # If tf not installed, import raises ImportError
+    import tensorflow as tf
+    if parse_version(tf.__version__) < parse_version(min_tf_version):
+        # TF pre-installed, but below the minimum required version
+        raise DeprecationWarning("TensorFlow version below minimum requirement")
+except (ImportError, DeprecationWarning) as e:
```

Comment:
In case of the deprecationwarning can we print the exception? So the user is informed that his (potential) gpu TF version is being updated?